packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -769,6 +769,19 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         return (currentActiveRequests() && getVotePhase() == Phase.Reveal);
     }
 
+    function _computePendingStakes(address wallet, uint256 amount) internal override {
+        if (getVotePhase() != Phase.Reveal) return;
+        uint256 currentRoundId = getCurrentRoundId();
+        // If we are during an active reveal phase then we store the pending stake for the roundId
+        for (uint256 i = 0; i < pendingPriceRequests.length; i = unsafe_inc(i)) {
+            if (_getRequestStatus(priceRequests[pendingPriceRequests[i]], currentRoundId) == RequestStatus.Active) {
+                _freezeRoundVariables(currentRoundId);
+                _setPendingStake(wallet, currentRoundId, amount);
+                break;
+            }
+        }
```

Comment:
Now that we're storing by round, do we still have to loop over all requests?

Also, why do we need to freeze the round variables? Can you explain what would happen if we didn't freeze them and waited for the first reveal for that?

Reply:
> Now that we're storing by round, do we still have to loop over all requests?

Good observation! This loop was doing the same check we do in `_inActiveReveal()` with the possibility to do something for every price request (storing the pending stake). 
Now that we only store the pending stake once for the current round, we can get rid of the loop and just call `_inActiveReveal()`. Thank you for spotting this! Fixed in the latest commits.

> Also, why do we need to freeze the round variables? Can you explain what would happen if we didn't freeze them and waited for the first reveal for that?

I've added comments to the code to explain why this is needed. The best way to explain this, as you suggested, is explaining what would happen if we don't freeze:

Suppose we don't  freeze and I stake 100 UMA during an active reveal phase **JUST BEFORE** any reveal is made and so before the `_freezeRoundVariables()` is called for this particular round. My pending stake will be incremented +100, my stake amount will be incremented +100 and the global cumulative stake as well, +100.

With the first reveal we run `_freezeRoundVariables()`, and then we store `rounds[roundId].cumulativeActiveStakeAtRound = cumulativeStake` this `cumulativeStake` includes my +100 UMA.

When the active reveal phase is over I call updateTrackers(Pablo). During the update for the price requests for the previous round we deduct the pendingStake to my stake:
`uint256 effectiveStake = voterStake.stake - voterStake.pendingStakes[priceRequest.lastVotingRound];`

But the `cumulativeActiveStakeAtRound` has my +100 UMA included (And all stakes of other users that occurred in the active reveal phase before the first reveal). There is an offset between the cumulativeActiveStakeAtRound and the sum of all the `effectiveStakes` of the voters, making the next calculations globally wrong. This results in a sum total of all slashes != 0.

This [afterEach](https://github.com/UMAprotocol/protocol/blob/e2181453ffb02f1efa10ce6d8e49baba584352b5/packages/core/test/oracle/VotingV2.js#L93) hook I added checks this balance, and we can verify how, by removing the `_freezeRoundVariables` from `_computePendingStakes`, there is a particular test in which the scenario I describe occurs, for which the sum is not 0 and fails.

Reply:
this is so clean now! great work @md0x